h/services/bulk_executor/_executor.py

```diff
@@ -0,0 +1,69 @@
+"""Bulk executor for use with h.h_api.bulk_api."""
+
+from sqlalchemy.orm.exc import NoResultFound
+
+from h.h_api.bulk_api.executor import AutomaticReportExecutor
+from h.h_api.enums import CommandType, DataType
+from h.h_api.exceptions import InvalidDeclarationError, UnsupportedOperationError
+from h.models import User
+
+
+class BulkExecutor(AutomaticReportExecutor):
+    """Executor of command objects which will modify the database in bulk."""
+
+    FAKE = object()
+
+    def __init__(self, db, authority="lms.hypothes.is"):
+        """
+        :param db: DB session object
+        :param authority: Restrict all request to this authority
+        """
+        self.db = db
+        self.authority = authority
+
+        self.handlers = {
+            (CommandType.UPSERT, DataType.USER): self.FAKE,
+            (CommandType.UPSERT, DataType.GROUP): self.FAKE,
+            (CommandType.CREATE, DataType.GROUP_MEMBERSHIP): self.FAKE,
```

Comment:
As the different parts get implemented, they can be slotted in here. Presently this just facilitates checking permissible combinations.